src/interpreter/mod.rs

```diff
@@ -1169,58 +1222,89 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 self.stack[frame].set_local(local, Value::ByVal(val));
                 Ok(())
             }
+            Lvalue::Static(cid) => {
+                let static_val = self.statics.get_mut(&cid).expect("static not cached");
+                assert!(static_val.mutable);
```

Comment:
Couldn't we reach this assert when a user erroneously attempts to write to a static? Then we would need to emit an error.